docs/federation.md

```diff
@@ -8,8 +8,12 @@ sort_rank: 6
 Federation allows a Prometheus server to scrape selected time series from
 another Prometheus server.
 
-_Note about native histograms (experimental feature): Federation does not
-support native histograms yet._
+Note about native histograms (experimental feature): Native histogram works only with
+protobuf scraping. When enabled, if the matchers result in a mix of float64 and histogram
+samples under the same metric name, the federation will return multiple metric families
+for the metric name. While this is technically not a valid exposition format, since the
+consumer is again a Prometheus server, Prometheus will accept multiple metric families
+for matching metric names.
```

Comment:
I guess we could specifically mention the histogram feature flag here rather than the technical detail of protobuf scraping. Suggestion (with some additional wordsmithing):

_Note about native histograms (experimental feature): To scrape native histograms via federation, the scraping Prometheus server needs to run with native histograms enabled (via the command line flag `--enable-feature=native-histograms`), implying that the protobuf format is used for scraping. Should the federated metrics contain a mix of different sample types (float64, counter histogram, gauge histogram) for the same metric name,  the federation payload will contain multiple metric families with the same name (but different types). Technically, this violates the rules of the protobuf exposition format, but Prometheus is nevertheless able to ingest all metrics correctly._